src/server/task-list/controller.js

```diff
@@ -13,63 +13,29 @@ const buttonText = 'Review and submit'
  */
 export const taskListGetController = {
   handler(req, h) {
-    const origin = Origin.fromState(req.yar.get('origin'))
-    const destination = Destination.fromState()
-    const tests = Tests.fromState()
-    const licence = Licence.fromState(req.yar.get('licence'))
-    const isOriginValid = origin.validate().isValid
-
-    const originGdsTask = buildGdsTaskItem({
-      title: 'Movement origin',
-      initialLink: 'origin/to-or-from-own-premises',
-      summaryLink: 'origin/summary',
-      isValid: isOriginValid,
-      isEnabled: true
-    })
-
-    const destinationGdsTask = buildGdsTaskItem({
-      title: 'Movement destination',
-      initialLink: '#',
-      summaryLink: '#',
-      isValid: destination.validate().isValid,
-      isEnabled: isOriginValid
-    })
+    const application = {
+      origin: Origin.fromState(req.yar.get('origin')),
+      destination: Destination.fromState(),
+      tests: Tests.fromState(),
+      licence: Licence.fromState(req.yar.get('licence'))
+    }
 
-    const testsGdsTask = buildGdsTaskItem({
-      title: 'Tests',
-      initialLink: '#',
-      summaryLink: '#',
-      isValid: tests.validate().isValid,
-      isEnabled: isOriginValid
-    })
+    const tasks = Object.values(application)
 
-    const licenceGdsTask = buildGdsTaskItem({
-      title: 'Receiving the licence',
-      initialLink: '/receiving-the-licence/licence-enter-email-address',
-      summaryLink: '/receiving-the-licence/check-answers',
-      isValid: licence.validate().isValid,
-      isEnabled: true
+    const gdsTasks = tasks.map((section) => {
+      return {
```

Comment:
should this be

```javascript
return buildGdsTaskItem({
  ...
})
```

The template expects something with a substantially different shape than we're passing in here.